pkg/device/volumeerror/volume_error.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volumeerror
```

Comment:
Do we need a separate volume error package. WDYT @iyashu ?

Reply:
I think, intention of introducing the typed error is to distinguish between different type of error code returned by provisioning (i.e its cause of insufficient capacity, invalid parameter or some error with execution of parted command). One improvement that I can think is to define it at crd level (similar to [what we did](https://github.com/openebs/lvm-localpv/blob/12ea20310329f68acec8db83c1d9b226101fd76e/pkg/apis/openebs.io/lvm/v1alpha1/lvmvolume.go#L108) for lvm), so that controller can also inspect it if required in future.

Reply:
@vakul shall we volume error type to crd itself, so that controller can also inspect the error kind if required in future. As an example, you can see [how we did](https://github.com/openebs/lvm-localpv/blob/12ea20310329f68acec8db83c1d9b226101fd76e/pkg/apis/openebs.io/lvm/v1alpha1/lvmvolume.go#L108) for lvm.

Reply:
resolved